### ISSUE TITLE
Outline view + Markdown: Do not display the "#"s

### ISSUE THREAD

**Issue by ssilva** [microsoft/vscode#53992]
The Outline view for Markdown already displays headers and sub-headers in a structured way; there is no need to pollute the view with "#"s to show the header levels.

![image](https://user-images.githubusercontent.com/103355/42536060-08a6d31e-845f-11e8-8846-4bb7a7fad80a.png)

**Comment by Stone-Zeng** [comment]
It would be better to use different icons (not only "[abc]") to show different header levels.

**Comment by odonyde** [comment]
I support this feature request: Displaying markdown hashes in the outline view is not only redundant (the tree structure already holds this information), but also distracts from the actual content (the section title).

**Comment by StefanoCecere** [comment]
we need a custom outline view just for markdown files

**Comment by ckosmowski** [comment]
It seems like this was done totally on purpose. Since i cannot estimate the side effects besides the display in the outline i won't file a merge request. But i think we could just remove `getSymbolName` and replace it with `entry.text` in two places here:

https://github.com/microsoft/vscode/blob/28bf0b1f56dc4530331edb2e6e8bbc3df192372e/extensions/markdown-language-features/src/features/documentSymbolProvider.ts#L55-L74

and then `getSymbolName` is also obsolete.

**Comment by pisces312** [comment]
I changed the getSymbolName function in resources\app\extensions\markdown-language-features\dist\extension.js to remove '#' char as a workaround.

``` js
getSymbolName(e){return" ".repeat(e.level)+e.text}
```

**Comment by koffeeboi** [comment]
This is what I've been using for a while.

I extended pisces312's idea with the following. 

1. Navigate to `%USERPROFILE%\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\markdown-language-features\dist` (Windows)
2. Replace the `getSymbolName` function with the following:
```js
getSymbolName(e){ var symbol = { "1": "⬜", "2": "🟥", "3": "🟧", "4": "🟨", "5": "🟩", "6": "🟦", "7": "🟪", "8": "🟫", }; if (1 <= e.level && e.level <= 8) { return symbol[e.level] + " " + e.text; } return "#".repeat(e.level) + " " + e.text; }
```

**Prettier version:**
```js
getSymbolName(e){
    var symbol = {
        "1": "⬜",
        "2": "🟥",
        "3": "🟧",
        "4": "🟨",
        "5": "🟩",
        "6": "🟦",
        "7": "🟪",
        "8": "🟫",
    };
    if (1 <= e.level && e.level <= 8) {
        return symbol[e.level] + " " + e.text;
    }
    return "#".repeat(e.level) + " " + e.text;
}
```

Screen shot of how it'll look like:
![image](https://user-images.githubusercontent.com/11512260/121842292-06ba5400-cc95-11eb-8cb7-69cf8cce9527.png)

**Comment by Zenahr** [comment]
@pisces312 answer should at least be implemented as a config option.

Here's a visual comparison showing how much more "clutter" the unnecessary `#`'s create:



![image](https://user-images.githubusercontent.com/47085752/125870853-b6c00612-9e18-482c-bbab-df31d57be58a.png)

![image](https://user-images.githubusercontent.com/47085752/125870951-020682c6-ec45-4314-89e1-bed3a3ff729d.png)

**Comment by yyangdid** [comment]
I also don't like the '`#` with outline, and I also don't want change the native source code file with  VSCode or extensions .

So, I temporarily use Emoji to better distinguish the headers level, this:

![Snipaste_2021-12-11_13-58-21](https://user-images.githubusercontent.com/10922667/145666013-701bcd0b-317c-484c-b01d-4b94c6b988b4.png)


And, I really want an option to hidden `#` 😁

**Comment by vancejc-mt** [comment]
Just ran across this - I wholeheartedly agree that this should be at least a setting; the hash chars don't add anything the indentation doesn't already address.

**Comment by Herman-Wu** [comment]
`extension.js` no longer has the `getSymbolName` function. Is there anyone who knows other way to remove the annoy "#"?  @pisces312 , @koffeeboi , @ckosmowski

**Comment by Herman-Wu** [comment]
After some file content search, I found now we need to modify `resources\app\extensions\markdown-language-features\server\dist\node\main.js` instead of `extension.js`. 
And the function name changes from `getSymbolName` to  `getTocSymbolName`. 
We can still use the original way to modify it.

**Comment by pisces312** [comment]
> After some file content search, I found now we need to modify `resources\app\extensions\markdown-language-features\server\dist\node\main.js` instead of `extension.js`. And the function name changes from `getSymbolName` to `getTocSymbolName`. We can still use the original way to modify it.

I just got the same issue and found you have figured out the solution. :)

**Comment by Acumane** [comment]
Would also be nice if we could hide syntax like \*\*bold**. It's just more noise if not actively styling text

**Comment by zyjdmmm** [comment]
Have we solved this problem？It's been many years。。。

**Comment by starball5** [comment]
Related question on Stack Overflow: [How to remove hash symbol "#" in outline of markdown?](https://stackoverflow.com/q/63570742/11107541)

**Comment by poa** [comment]
From v1.76.0 file is: `resources\app\extensions\markdown-language-features\server\dist\node\workerMain.js`
Function: `getTocSymbolName`

**Comment by AlbertoFabbri93** [comment]
I agree that the hash chars add unnecessary clutter to the outline view. It would be awesome to have a setting for this

**Comment by nonplayer** [comment]
This is especially frustrating as a Portable Mode user. Having to go in and manually make this change every time I upgrade VSCode is a chore.

**Comment by nonplayer** [comment]
May 2024 and this is still an issue with Markdown All in One 3.6.2 and VSCode 1.89.0. Seems the function has again changed names. It's no longer `#getTocSymbolName(e)` now it's just `#se(e)` - ignoring the function name, just searching the workerMain.js file for:

```
{return"#".repeat(e.level)+" "+e.text}
```

and replacing it with:

```
{return" ".repeat(e.level)+e.text}
```
 still applies the fix.

**Comment by jim0203** [comment]
I'm using 1.92.1 on macOS. What would be the path and file that is equivalent to `resources\app\extensions\markdown-language-features\server\dist\node\workerMain.js`?

I've found this path:

`/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/markdown-language-features/dist`

Which contains a `serverWorkerMain.js` file. That file includes the text that @nonplayer says should be replaced, but replacing it doesn't fix the issue.

Do I need to do something else (beyond restarting VSCode) to get the change to register? Or am I editing the wrong file?

**Comment by AfroThundr3007730** [comment]
For those who may be using a remote development extension, you can make the above edit using the below code (on the remote):

```bash
shopt -s globstar
# It doesn't matter where you hide, sed will find you...
sed -i 's/return"#".repeat(e.level)+" "+e.text/return e.text/g' \
    ~/.vscode-server/cli/servers/Stable-*/server/extensions/markdown-language-features/**/*.js
```

**Comment by leichau** [comment]
It is obvious that this is a problem that directly affects the user experience. Moreover, this issue appears to be easily solvable.

Why hasn't such a small issue been resolved for 7 years, even though a small modification can greatly improve the user experience.

From the above discussion, it can be seen that many users have to manually correct this issue every time they update.

If VSCode truly values user experience, please resolve this issue as soon as possible.

I don't know how to make you pay attention to this situation, so I have to @ possible relevant developers.

@kieferrm @mjbvz @amunger @aeschli @iwangbowen @alexdima @rebornix

**Comment by oeloo** [comment]
I agree that the `##` in the Outline View should not be visible in the VS Code Outline View.
`##` is a formatting marker for .md files for telling that this is a header, so it does not have its place when the header is represented as a header.

Unfortunately, this issue has remained unresolved since 2018.
Is there a VS Code extension extending the Outline View? The Outline view is a crucial Markdown feature that requires full attention.

**Comment by valkalon** [comment]
if you could add the setting "Outline: Show markdown #" with a simple checkbox, that would be fantastic.

**Comment by sck-jenzabar** [comment]
I would also like to see the outline work when viewing the Markdown preview full screen. If this is also important to you, please consider upvoting this related issue: https://github.com/microsoft/vscode/issues/316113

**Comment by nonplayer** [comment]
It's 2026, and this has been open now for eight years. In fact, in recent version 1.125.0 they even overhauled the entire `serverWorkerMain.js` file and changed the code... and *still* someone deliberately chose to keep this bug in place.

The new snippet to search for is: `return "#".repeat(entry.level) + " " + entry.text;`

Replace it with: `return "".repeat(entry.level) + entry.text;`

**Comment by valkalon** [comment]
thanks @nonplayer . Yes would be great to solve this finally.
